library/core/src/option.rs

```diff
@@ -162,6 +162,8 @@ pub enum Option<T> {
     /// No value
     #[lang = "None"]
     #[stable(feature = "rust1", since = "1.0.0")]
+    #[doc(alias = "null")]
+    #[doc(alias = "nil")]
```

Comment:
C does not have `nil`, and `null` should be `NULL` if it is referring to the library side of C.

Reply:
I can separate this into a separate PR if you like, but I think the aliases are useful.

> null should be NULL if it is referring to the library side of C.

`doc(alias)` is case insensitive.

Reply:
Still, writing the original name as it is supposed to be would be better, no? Otherwise we should change e.g. `Map` to `map` in the Java one, etc.